From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Venelin <venelin@pulumi.com>
Date: Fri, 3 Jan 2025 16:37:43 +0200
Subject: [PATCH] rename input property on guardrails to avoid name clash


diff --git a/internal/service/serving_tf/legacy_model.go b/internal/service/serving_tf/legacy_model.go
index 80e01fa1..cf5f9619 100755
--- a/internal/service/serving_tf/legacy_model.go
+++ b/internal/service/serving_tf/legacy_model.go
@@ -632,7 +632,7 @@ func (m AiGatewayGuardrailPiiBehavior_SdkV2) Type(ctx context.Context) attr.Type
 
 type AiGatewayGuardrails_SdkV2 struct {
 	// Configuration for input guardrail filters.
-	Input types.List `tfsdk:"input"`
+	Input types.List `tfsdk:"input_properties"`
 	// Configuration for output guardrail filters.
 	Output types.List `tfsdk:"output"`
 }
@@ -678,8 +678,8 @@ func (to *AiGatewayGuardrails_SdkV2) SyncFieldsDuringRead(ctx context.Context, f
 }
 
 func (m AiGatewayGuardrails_SdkV2) ApplySchemaCustomizations(attrs map[string]tfschema.AttributeBuilder) map[string]tfschema.AttributeBuilder {
-	attrs["input"] = attrs["input"].SetOptional()
-	attrs["input"] = attrs["input"].(tfschema.ListNestedAttributeBuilder).AddValidator(listvalidator.SizeAtMost(1)).(tfschema.AttributeBuilder)
+	attrs["input_properties"] = attrs["input_properties"].SetOptional()
+	attrs["input_properties"] = attrs["input_properties"].(tfschema.ListNestedAttributeBuilder).AddValidator(listvalidator.SizeAtMost(1)).(tfschema.AttributeBuilder)
 	attrs["output"] = attrs["output"].SetOptional()
 	attrs["output"] = attrs["output"].(tfschema.ListNestedAttributeBuilder).AddValidator(listvalidator.SizeAtMost(1)).(tfschema.AttributeBuilder)
 
@@ -695,8 +695,8 @@ func (m AiGatewayGuardrails_SdkV2) ApplySchemaCustomizations(attrs map[string]tf
 // SDK values.
 func (m AiGatewayGuardrails_SdkV2) GetComplexFieldTypes(ctx context.Context) map[string]reflect.Type {
 	return map[string]reflect.Type{
-		"input":  reflect.TypeOf(AiGatewayGuardrailParameters_SdkV2{}),
-		"output": reflect.TypeOf(AiGatewayGuardrailParameters_SdkV2{}),
+		"input_properties": reflect.TypeOf(AiGatewayGuardrailParameters_SdkV2{}),
+		"output":           reflect.TypeOf(AiGatewayGuardrailParameters_SdkV2{}),
 	}
 }
 
@@ -707,8 +707,8 @@ func (m AiGatewayGuardrails_SdkV2) ToObjectValue(ctx context.Context) basetypes.
 	return types.ObjectValueMust(
 		m.Type(ctx).(basetypes.ObjectType).AttrTypes,
 		map[string]attr.Value{
-			"input":  m.Input,
-			"output": m.Output,
+			"input_properties": m.Input,
+			"output":           m.Output,
 		})
 }
 
@@ -716,7 +716,7 @@ func (m AiGatewayGuardrails_SdkV2) ToObjectValue(ctx context.Context) basetypes.
 func (m AiGatewayGuardrails_SdkV2) Type(ctx context.Context) attr.Type {
 	return types.ObjectType{
 		AttrTypes: map[string]attr.Type{
-			"input": basetypes.ListType{
+			"input_properties": basetypes.ListType{
 				ElemType: AiGatewayGuardrailParameters_SdkV2{}.Type(ctx),
 			},
 			"output": basetypes.ListType{
@@ -748,7 +748,7 @@ func (m *AiGatewayGuardrails_SdkV2) GetInput(ctx context.Context) (AiGatewayGuar
 // SetInput sets the value of the Input field in AiGatewayGuardrails_SdkV2.
 func (m *AiGatewayGuardrails_SdkV2) SetInput(ctx context.Context, v AiGatewayGuardrailParameters_SdkV2) {
 	vs := []attr.Value{v.ToObjectValue(ctx)}
-	t := m.Type(ctx).(basetypes.ObjectType).AttrTypes["input"]
+	t := m.Type(ctx).(basetypes.ObjectType).AttrTypes["input_properties"]
 	m.Input = types.ListValueMust(t, vs)
 }
 
diff --git a/internal/service/serving_tf/model.go b/internal/service/serving_tf/model.go
index 243fb875..c9338bd3 100755
--- a/internal/service/serving_tf/model.go
+++ b/internal/service/serving_tf/model.go
@@ -611,7 +611,7 @@ func (m AiGatewayGuardrailPiiBehavior) Type(ctx context.Context) attr.Type {
 
 type AiGatewayGuardrails struct {
 	// Configuration for input guardrail filters.
-	Input types.Object `tfsdk:"input"`
+	Input types.Object `tfsdk:"input_properties"`
 	// Configuration for output guardrail filters.
 	Output types.Object `tfsdk:"output"`
 }
@@ -657,7 +657,7 @@ func (to *AiGatewayGuardrails) SyncFieldsDuringRead(ctx context.Context, from Ai
 }
 
 func (m AiGatewayGuardrails) ApplySchemaCustomizations(attrs map[string]tfschema.AttributeBuilder) map[string]tfschema.AttributeBuilder {
-	attrs["input"] = attrs["input"].SetOptional()
+	attrs["input_properties"] = attrs["input_properties"].SetOptional()
 	attrs["output"] = attrs["output"].SetOptional()
 
 	return attrs
@@ -672,8 +672,8 @@ func (m AiGatewayGuardrails) ApplySchemaCustomizations(attrs map[string]tfschema
 // SDK values.
 func (m AiGatewayGuardrails) GetComplexFieldTypes(ctx context.Context) map[string]reflect.Type {
 	return map[string]reflect.Type{
-		"input":  reflect.TypeOf(AiGatewayGuardrailParameters{}),
-		"output": reflect.TypeOf(AiGatewayGuardrailParameters{}),
+		"input_properties": reflect.TypeOf(AiGatewayGuardrailParameters{}),
+		"output":           reflect.TypeOf(AiGatewayGuardrailParameters{}),
 	}
 }
 
@@ -684,8 +684,8 @@ func (m AiGatewayGuardrails) ToObjectValue(ctx context.Context) basetypes.Object
 	return types.ObjectValueMust(
 		m.Type(ctx).(basetypes.ObjectType).AttrTypes,
 		map[string]attr.Value{
-			"input":  m.Input,
-			"output": m.Output,
+			"input_properties": m.Input,
+			"output":           m.Output,
 		})
 }
 
@@ -693,8 +693,8 @@ func (m AiGatewayGuardrails) ToObjectValue(ctx context.Context) basetypes.Object
 func (m AiGatewayGuardrails) Type(ctx context.Context) attr.Type {
 	return types.ObjectType{
 		AttrTypes: map[string]attr.Type{
-			"input":  AiGatewayGuardrailParameters{}.Type(ctx),
-			"output": AiGatewayGuardrailParameters{}.Type(ctx),
+			"input_properties": AiGatewayGuardrailParameters{}.Type(ctx),
+			"output":           AiGatewayGuardrailParameters{}.Type(ctx),
 		},
 	}
 }
