From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Venelin <venelin@pulumi.com>
Date: Fri, 3 Jan 2025 16:37:43 +0200
Subject: [PATCH] rename input property on guardrails to avoid name clash


diff --git a/internal/service/serving_tf/legacy_model.go b/internal/service/serving_tf/legacy_model.go
index 8835a1f5..fab5e51d 100755
--- a/internal/service/serving_tf/legacy_model.go
+++ b/internal/service/serving_tf/legacy_model.go
@@ -591,7 +591,7 @@ func (o AiGatewayGuardrailPiiBehavior_SdkV2) Type(ctx context.Context) attr.Type
 
 type AiGatewayGuardrails_SdkV2 struct {
 	// Configuration for input guardrail filters.
-	Input types.List `tfsdk:"input"`
+	Input types.List `tfsdk:"input_properties"`
 	// Configuration for output guardrail filters.
 	Output types.List `tfsdk:"output"`
 }
@@ -635,8 +635,8 @@ func (toState *AiGatewayGuardrails_SdkV2) SyncFieldsDuringRead(ctx context.Conte
 }
 
 func (c AiGatewayGuardrails_SdkV2) ApplySchemaCustomizations(attrs map[string]tfschema.AttributeBuilder) map[string]tfschema.AttributeBuilder {
-	attrs["input"] = attrs["input"].SetOptional()
-	attrs["input"] = attrs["input"].(tfschema.ListNestedAttributeBuilder).AddValidator(listvalidator.SizeAtMost(1)).(tfschema.AttributeBuilder)
+	attrs["input_properties"] = attrs["input_properties"].SetOptional()
+	attrs["input_properties"] = attrs["input_properties"].(tfschema.ListNestedAttributeBuilder).AddValidator(listvalidator.SizeAtMost(1)).(tfschema.AttributeBuilder)
 	attrs["output"] = attrs["output"].SetOptional()
 	attrs["output"] = attrs["output"].(tfschema.ListNestedAttributeBuilder).AddValidator(listvalidator.SizeAtMost(1)).(tfschema.AttributeBuilder)
 
@@ -652,7 +652,7 @@ func (c AiGatewayGuardrails_SdkV2) ApplySchemaCustomizations(attrs map[string]tf
 // SDK values.
 func (a AiGatewayGuardrails_SdkV2) GetComplexFieldTypes(ctx context.Context) map[string]reflect.Type {
 	return map[string]reflect.Type{
-		"input":  reflect.TypeOf(AiGatewayGuardrailParameters_SdkV2{}),
+		"input_properties":  reflect.TypeOf(AiGatewayGuardrailParameters_SdkV2{}),
 		"output": reflect.TypeOf(AiGatewayGuardrailParameters_SdkV2{}),
 	}
 }
@@ -664,7 +664,7 @@ func (o AiGatewayGuardrails_SdkV2) ToObjectValue(ctx context.Context) basetypes.
 	return types.ObjectValueMust(
 		o.Type(ctx).(basetypes.ObjectType).AttrTypes,
 		map[string]attr.Value{
-			"input":  o.Input,
+			"input_properties":  o.Input,
 			"output": o.Output,
 		})
 }
@@ -673,7 +673,7 @@ func (o AiGatewayGuardrails_SdkV2) ToObjectValue(ctx context.Context) basetypes.
 func (o AiGatewayGuardrails_SdkV2) Type(ctx context.Context) attr.Type {
 	return types.ObjectType{
 		AttrTypes: map[string]attr.Type{
-			"input": basetypes.ListType{
+			"input_properties": basetypes.ListType{
 				ElemType: AiGatewayGuardrailParameters_SdkV2{}.Type(ctx),
 			},
 			"output": basetypes.ListType{
@@ -705,7 +705,7 @@ func (o *AiGatewayGuardrails_SdkV2) GetInput(ctx context.Context) (AiGatewayGuar
 // SetInput sets the value of the Input field in AiGatewayGuardrails_SdkV2.
 func (o *AiGatewayGuardrails_SdkV2) SetInput(ctx context.Context, v AiGatewayGuardrailParameters_SdkV2) {
 	vs := []attr.Value{v.ToObjectValue(ctx)}
-	t := o.Type(ctx).(basetypes.ObjectType).AttrTypes["input"]
+	t := o.Type(ctx).(basetypes.ObjectType).AttrTypes["input_properties"]
 	o.Input = types.ListValueMust(t, vs)
 }
 
diff --git a/internal/service/serving_tf/model.go b/internal/service/serving_tf/model.go
index 89afd705..1e0665ab 100755
--- a/internal/service/serving_tf/model.go
+++ b/internal/service/serving_tf/model.go
@@ -570,7 +570,7 @@ func (o AiGatewayGuardrailPiiBehavior) Type(ctx context.Context) attr.Type {
 
 type AiGatewayGuardrails struct {
 	// Configuration for input guardrail filters.
-	Input types.Object `tfsdk:"input"`
+	Input types.Object `tfsdk:"input_properties"`
 	// Configuration for output guardrail filters.
 	Output types.Object `tfsdk:"output"`
 }
@@ -614,7 +614,7 @@ func (toState *AiGatewayGuardrails) SyncFieldsDuringRead(ctx context.Context, fr
 }
 
 func (c AiGatewayGuardrails) ApplySchemaCustomizations(attrs map[string]tfschema.AttributeBuilder) map[string]tfschema.AttributeBuilder {
-	attrs["input"] = attrs["input"].SetOptional()
+	attrs["input_properties"] = attrs["input_properties"].SetOptional()
 	attrs["output"] = attrs["output"].SetOptional()
 
 	return attrs
@@ -629,7 +629,7 @@ func (c AiGatewayGuardrails) ApplySchemaCustomizations(attrs map[string]tfschema
 // SDK values.
 func (a AiGatewayGuardrails) GetComplexFieldTypes(ctx context.Context) map[string]reflect.Type {
 	return map[string]reflect.Type{
-		"input":  reflect.TypeOf(AiGatewayGuardrailParameters{}),
+		"input_properties":  reflect.TypeOf(AiGatewayGuardrailParameters{}),
 		"output": reflect.TypeOf(AiGatewayGuardrailParameters{}),
 	}
 }
@@ -641,7 +641,7 @@ func (o AiGatewayGuardrails) ToObjectValue(ctx context.Context) basetypes.Object
 	return types.ObjectValueMust(
 		o.Type(ctx).(basetypes.ObjectType).AttrTypes,
 		map[string]attr.Value{
-			"input":  o.Input,
+			"input_properties":  o.Input,
 			"output": o.Output,
 		})
 }
@@ -650,7 +650,7 @@ func (o AiGatewayGuardrails) ToObjectValue(ctx context.Context) basetypes.Object
 func (o AiGatewayGuardrails) Type(ctx context.Context) attr.Type {
 	return types.ObjectType{
 		AttrTypes: map[string]attr.Type{
-			"input":  AiGatewayGuardrailParameters{}.Type(ctx),
+			"input_properties":  AiGatewayGuardrailParameters{}.Type(ctx),
 			"output": AiGatewayGuardrailParameters{}.Type(ctx),
 		},
 	}
